### PR TITLE
Twilio hooks refactor

### PR DESCRIPTION
**Pull request by dmitrymarchuk**
Twilio rooms & participants management code was duplicated in a couple of places. This PR ensures that there is only one component that render user's video feed - `VideoParticipant`. And there is only one hook that connects to the room, listens for participants etc - `useVideoRoomState`.
Also:

- `LocalParticipant` and `Participant` components are completely removed
- all usages of twilio related stuff are removed from various places (`Room.tsx` etc.)
- a bunch of renames
- CSS fixes for new components to render correctly
- support for remote participants' audit and video mute&unmute in new code (was missing before)

### CODE REVIEW

File: src/components/organisms/Room/Room.tsx [hunk 1]
```diff
@@ -155,18 +56,24 @@ const Room: React.FC<RoomProps> = ({
...
+        className={`participant-container ${participantContainerClassName}`}
+        key={localParticipant.identity}
+      >
+        <VideoParticipant
+          participant={localParticipant}
+          participantUser={userWithId}
           defaultMute={defaultMute}
         />
       </div>
     ) : null;
```
Comment (mike-lvov) [inline src/components/organisms/Room/Room.tsx:70]:
```suggestion
    return localParticipant && userWithId &&
      <div
        className={`participant-container ${participantContainerClassName}`}
        key={localParticipant.identity}
      >
        <VideoParticipant
          participant={localParticipant}
          participantUser={userWithId}
          defaultMute={defaultMute}
        />
      </div>
```

File: src/components/organisms/Room/Room.tsx [hunk 2]
```diff
@@ -210,24 +116,18 @@ const Room: React.FC<RoomProps> = ({
...
-      <VideoErrorModal
-        show={!!videoError}
-        onHide={() => setVideoError("")}
-        errorMessage={videoError}
-        onRetry={connectToVideoRoom}
-        onBack={() => (setSeatedAtTable ? leaveSeat() : setVideoError(""))}
-      />
+      {renderErrorModal((dismissVideoError: () => void) =>
+        setSeatedAtTable ? leaveSeat() : dismissVideoError
+      )}
```
Comment (mike-lvov) [inline src/components/organisms/Room/Room.tsx:130]: It looks junky. Not only it is located in `return`, but it also has strange logic. If bool is true, call a function, if not - return a function

File: src/components/templates/FireBarrel/FireBarrel.tsx [hunk 3]
```diff
@@ -24,10 +23,11 @@ export interface FireBarrelProps {
...
 
-  const { room, participants } = useVideoRoomState({
-    user: userWithId,
-    roomName: venue?.name,
-  });
+  const {
+    localParticipant,
+    participants,
+    renderErrorModal,
+  } = useVideoRoomState(userId, venue?.name);
```
Comment (mike-lvov) [inline src/components/templates/FireBarrel/FireBarrel.tsx:30]: venue name, not ID?

File: src/hooks/twilio/useParticipantMediaState.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  useEffect(() => {
+    if (isEnabled) {
+      tracks.filter(isLocalTrack).forEach((localTrack) => localTrack.enable());
+    } else {
+      tracks.filter(isLocalTrack).forEach((localTrack) => localTrack.disable());
+    }
+  }, [isEnabled, tracks]);
+
+  // @debt should we be handling the other tracks too?
+  const track = tracks[0];
```
Comment (mike-lvov) [inline src/hooks/twilio/useParticipantMediaState.ts:119]:
```suggestion
  const [firstTrack] = tracks;
```

**Review by mike-lvov — COMMENTED** [review]
Did a go-through. It's quite large. Did dive deeply into the hooks refactor

File: src/components/organisms/Room/Room.tsx (hunk 1 above)
Comment (dmitrymarchuk) [reply src/components/organisms/Room/Room.tsx]: fixed in https://github.com/sparkletown/sparkle/pull/2369/commits/8fa06206d8a463463d85380f61fff2b4dd953adb

File: src/components/organisms/Room/Room.tsx (hunk 2 above)
Comment (dmitrymarchuk) [reply src/components/organisms/Room/Room.tsx]: fixed in https://github.com/sparkletown/sparkle/pull/2369/commits/8fa06206d8a463463d85380f61fff2b4dd953adb

File: src/components/templates/FireBarrel/FireBarrel.tsx (hunk 3 above)
Comment (dmitrymarchuk) [reply src/components/templates/FireBarrel/FireBarrel.tsx:30]: Yep, that's how it worked before refactoring. Changed it to use venue.id everywhere in https://github.com/sparkletown/sparkle/pull/2369/commits/e3fb396a321fe5ad2e70c112614770f643c07016 as its seems more robust.

File: src/hooks/twilio/useParticipantMediaState.ts (hunk 4 above)
Comment (dmitrymarchuk) [reply src/hooks/twilio/useParticipantMediaState.ts]: fixed in https://github.com/sparkletown/sparkle/pull/2369/commits/02f1bf1b6515deb73d5fbafef179603f154478e7

**Review by mike-lvov — APPROVED** [review]
Awesome job!

**Pull request merged**